File: GameInsightsApp.Models/Entities/GameEvent.cs
using System;
using System.Collections.Generic;
using System.Text;
using GameInsightsApp.Models.Enums;

namespace GameInsightsApp.Models.Entities
{
    public class GameEvent
    {
        public int Id { get; set; }
        public int GameId { get; set; }
        public int? PlayerId { get; set; }
        public int? TeamId { get; set; }
        public GameEventType EventType { get; set; }
        public string Notes { get; set; }
    }
}

File: GameInsightsApp/ClientApp/src/app/models/game-event.ts
export interface GameEvent {
  id: number;
  gameId: number;
  playerId?: number;
  teamId?: number;
  gameEventType: GameEventType;
  notes: string;
}

export enum GameEventType {
  CapturedTheFlag = 0,
  FlagPickedUp,
  LostTheFlag,
  FlagInterceptedByAnotherTeam,
  FlagIsReturnedToTheBase,
  FlagIsReturnedToTheBaseByTimeout,
  FlagIsBroughtToTheEnemyBase
}
